### DISCUSSION TITLE
Plotting Layers - Getting outline layer

### DISCUSSION THREAD

**Discussion by ghost** [ansys/pyaedt#1720] [Q&A] [answered]
In [EDBLayer.plot](https://aedt.docs.pyansys.com/API/_autosummary/pyaedt.edb_core.EDB_Data.EDBLayer.plot.html) there is an optional outline parameter. I would like to plot my selected layer with the board outline in the image. I have tried the [EDBLayers.add_outline_layer](https://aedt.docs.pyansys.com/API/_autosummary/pyaedt.edb_core.EDB_Data.EDBLayers.add_outline_layer.html) function but it will return false with the default arguments. This would suggest there is already a layer named "Outline" in the project. However, I'm not sure how to access the outline layer... and how to get it into the list form required for the EDBLayer.plot method.
In addition, it seems there is no way to edit the plot title, axes; is the preferred way to handle this with the [EDBNets.get_plot_data](https://aedt.docs.pyansys.com/API/_autosummary/pyaedt.edb_core.nets.EdbNets.get_plot_data.html) method? Could the EDBLayer.plot method be modified to take the standard matplotlib arguments for title, axis labels, etc?

**Comment by svandenb-dev** [comment, marked as answer]
Hi @pyAnsysUser,
The outline layer is a non stackup one and is always included in an EDB database. To access it:
edb.stackup.non_stackup_layers["Outline"].
However the plot command is only valid for physical layers e.g stackup ones.
